src/player.cpp

```diff
@@ -9831,6 +9832,37 @@ bool player::can_use( const item& it, bool interactive ) const {
     });
 }
 
+bool player::can_reload( const item& it, const itype_id& ammo ) const {
+    if( !it.is_reloadable() ) {
+        return false;
+    }
+
+    if( it.is_ammo_belt() ) {
+        auto linkage = it.type->magazine->linkage;
+        if( linkage != "NULL" && !has_charges( linkage, 1 ) ) {
+            return false;
+        }
+    }
+
+    if( it.magazine_integral() ) {
+        if( !ammo.empty() ) {
+            if( it.ammo_data() ) {
+                if( it.ammo_data()->id != ammo ) {
+                    return false;
+                }
+            } else {
+                auto at = item_controller->find_template( ammo );
```

Comment:
You used `item::find_type` in the new iuse_actor, why not here? That way you don't need to include item_factory.h